MyToolsYourToolsBackend/MyToolsYourToolsBackend.Application/Services/UserGroupService.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using AutoMapper;
+using MyToolsYourToolsBackend.Application.Dtos;
+using MyToolsYourToolsBackend.Domain.DbContexts;
+using MyToolsYourToolsBackend.Domain.Entities;
+
+namespace MyToolsYourToolsBackend.Application.Services
+{
+    public class UserGroupService : IUserGroupService
+    {
+        private AppDbContext _dbContext;
+
+        public UserGroupService(AppDbContext dbContext)
+        {
+            _dbContext = dbContext;
+        }
+       
+        public UserGroupDto JoinGroup(UserGroupDto userGroup)
+        {
+            var userGroupToSave = Mapper.Map<UserGroup>(userGroup);
+
+            _dbContext.UserGroups.Add(userGroupToSave);
+
+            if (_dbContext.SaveChanges() == 0)
+            {
+                throw new Exception("Could not join the group");
+            }
+
+            return userGroup;
+        }
+
+        public bool LeaveGroup(UserGroupDto userGroup)
+        {
+            var userGroupToRemove = _dbContext.UserGroups.Find(new object[] { userGroup.UserId, userGroup.GroupId });
```

Comment:
chyba lepiej firstOrDefualt(ug => ug.UserId  == userGroup.UserId && ug.GroupId == userGroup.GroupId)

Reply:
Właśnie zastanawiałem się czemu tego używasz, bo co to jest to default? Bałem się że mi coś będzie podstawiać jak nie znajdzie, chciałem mieć pewność ze jak nie ma to wywali wyjątek. Wiesz czy to się różnie od Find()?

Reply:
firstOrDefult zwraca pierwsza wartosc do której pasuje lambda lub nulla jesli tego nie ma. No te find tez zadziała ale jak widzisz musisz tworzyć jakiś anonimowy obiekt i to może własnie w przyszłości jakies trudne do znalezienia błedy wytworzyć (np. jak zmienimy klase user group czy cos)